cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/transact.rs

```diff
@@ -0,0 +1,249 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{
+	create_pool_with_native_on,
+	tests::{snowbridge::CHAIN_ID, *},
+};
+use sp_core::Get;
+use xcm::latest::AssetTransferFilter;
+
+const ETHEREUM_BOB: [u8; 20] = hex_literal::hex!("11b0b11000011b0b11000011b0b11000011b0b11");
+
+/// Bob on Ethereum transacts on PenpalB, paying fees using WETH. XCM has to go through Asset Hub
+/// as the reserve location of WETH. The original origin `Ethereum/Bob` is proxied by Asset Hub.
+///
+/// This particular test is not testing snowbridge, but only Bridge Hub, so the tested XCM flow from
+/// Ethereum starts from Bridge Hub.
+// TODO(https://github.com/paritytech/polkadot-sdk/issues/6243): Once Snowbridge supports Transact, start the flow from Ethereum and test completely e2e.
+fn transfer_and_transact_in_same_xcm(
+	sender: Location,
+	weth: Asset,
+	destination: Location,
+	beneficiary: Location,
+	call: xcm::DoubleEncoded<()>,
+) {
+	let signed_origin = <BridgeHubWestend as Chain>::RuntimeOrigin::root();
+	let context: InteriorLocation = [
+		GlobalConsensus(Westend),
+		Parachain(<BridgeHubWestend as Para>::ParachainInfo::get().into()),
+	]
+	.into();
+	let asset_hub_location = BridgeHubWestend::sibling_location_of(AssetHubWestend::para_id());
+
+	// TODO(https://github.com/paritytech/polkadot-sdk/issues/6197): dry-run to get local fees, for now use hardcoded value.
+	let ah_fees_amount = 90_000_000_000u128; // current exact value 79_948_099_299
+	let fees_for_ah: Asset = (weth.id.clone(), ah_fees_amount).into();
+
+	let require_weight_at_most = Weight::from_parts(1000000000, 200000);
+	// xcm to be executed at dest
+	let xcm_on_dest = Xcm(vec![
+		Transact { origin_kind: OriginKind::Xcm, require_weight_at_most, call },
```

Comment:
`require_weight_at_most` is deprecated.

Reply:
Yeah, still haven't integrated that here it seems